### PR TITLE
Add setuptools to test dependencies

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description
Two functional tests import `setuptools`. I noticed these failures when creating a new Python 3.12 environment and following the contributor instructions:

```py
____ test_functional[recursion_error_3152] ____

self = <pylint.testutils.lint_module_test.LintModuleTest object at 0x12c807f80>

    def runTest(self) -> None:
>       self._runTest()
E       AssertionError: Wrong message(s) raised for "recursion_error_3152.py":
E       
E       Unexpected in testdata:
E          2: import-error

pylint/testutils/lint_module_test.py:147: AssertionError
____ test_functional[recursion_error_3159] ____

self = <pylint.testutils.lint_module_test.LintModuleTest object at 0x12c849be0>

    def runTest(self) -> None:
>       self._runTest()
E       AssertionError: Wrong message(s) raised for "recursion_error_3159.py":
E       
E       Unexpected in testdata:
E          6: import-error

pylint/testutils/lint_module_test.py:147: AssertionError
```

### CODE REVIEW

**Comment by Pierre-Sassoulas** [comment]
Maybe we could import from pytest instead ?

**Comment by jacobtylerwalls** [comment]
The two regressions tests seem specific to `setuptools`, but of course this will change over time as those packages change:

https://github.com/pylint-dev/pylint/issues/3152
https://github.com/pylint-dev/pylint/issues/3159

**Comment by jacobtylerwalls** [comment]
But the [fix](https://github.com/pylint-dev/astroid/commit/ed42e0ceea0d2b1eb9cbac3562df28a185c85aaf) doesn't have anything to do with setuptools 🤔 ...

**Comment by Pierre-Sassoulas** [comment]
Testing with setuptools have no effect anymore though, rmaking the pylint test suite test the original bug again would require a bigger change (creating a reproducer by using the old setuptools code?)

**Comment by jacobtylerwalls** [comment]
> Testing with setuptools have no effect anymore though

My bad, but this is the part I don't understand. It would prevent `import-error` for these two tests.

> (creating a reproducer by using the old setuptools code?)

We don't know the setuptools code has actually changed. It might be that we protected against the recursion issue by some other method along the way.

File: requirements_test_min.txt [hunk 1]
```diff
@@ -8,3 +8,4 @@ pytest-benchmark~=4.0
 pytest-timeout~=2.1
 towncrier~=23.6
 requests
+setuptools
```
Comment (Pierre-Sassoulas) [inline requirements_test_min.txt:11]: That makes sense, what about fixing the setuptools version to the old code that was live during #3152. That's october 2019, so :
```suggestion
setuptools==41.6.0
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]

File: requirements_test_min.txt (hunk 1 above)
Comment (jacobtylerwalls) [reply requirements_test_min.txt:11]: I guess what I don't love about that is that you'd need separate environments for testing and building. Could cause confusion.

Comment (Pierre-Sassoulas) [reply requirements_test_min.txt:11]: Do we ever build ? I  was at some point but now the release is done by a github action and I never build locally (or tox build locally?)

Comment (jacobtylerwalls) [reply requirements_test_min.txt:11]: Fair point!

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**